server/ntb/commands/sync_topics.py

```diff
@@ -0,0 +1,402 @@
+        """Generate data for reporting the changes to the local Media Topics CV
+
+        ``report["new"]``: New items from IPTC that aren't in local CV
+        ``report["deviated"]``: List of items where attribute values have deviated from IPTC
+        """
+
+        report = {
+            "new": [],
+            "deviated": [],
+        }
+
+        for topic in updated_topics:
+            original = existing_topics.get(topic["qcode"])
+            topic["_existing"] = original
+
+            if not original:
+                # This is a new Media Topic, add it to the report
+                report["new"].append(topic)
+                continue
+            elif not original.get("is_active"):
+                # Skip this Media Topic, as it is disabled in local CV
+                continue
+
+            deviations = []
+
+            for field in ["name", "parent", "wikidata", "iptc_subject"]:
+                if (
+                    len(original.get(field) or "") and
+                    len(topic.get(field) or "") and
+                    topic[field] != original[field]
+
+                ):
+                    # This field exists in both local CV and IPTC
+                    # but the values differ. Add this to list of deviated fields
+                    deviations.append(field)
+
+            if len(deviations):
+                # The local Media Topic has deviated from IPTC
+                # Add this to the report
+                report["deviated"].append({
+                    "topic": topic,
+                    "fields": deviations,
+                })
+
+        return report
+
+    def _generate_report(self, report, topics):
+        """Generate and write the report to a markdown file
+
+        Writes the file to ``server/ntb/commands/syn_topics_reports``
+        The report uses markdown and table to show the data.
+        """
+
+        output = self._gen_new_table(report["new"])
+
+        output += self._gen_translation_table(topics)
+        output += self._gen_deviated_table(report["deviated"])
+
+        file_path = path.abspath(
+            path.join(
+                path.dirname(__file__),
+                "sync_topics_reports",
+                f"{datetime.now():%Y-%m-%dT%H-%M}.md"
+            )
+        )
+
+        with open(file_path, "w") as f:
+            f.write(output)
+
+    def _gen_new_table(self, topics):
+        """Generate markdown table for newly added IPTC Media Topics"""
+
+        report = """
+New Items:
+
+| Qcode | Name | Parent | Wikidata | IPTC Subject |
+| ----- | ---- | ------ | -------- | ------------ |
+"""
+        for topic in topics:
+            qcode = topic.get("qcode") or ""
+            name = topic.get("name") or ""
+            parent = topic.get("parent") or ""
+            wikidata = topic.get("wikidata") or ""
+            iptc_subject = topic.get("iptc_subject") or ""
+            report += f"| {qcode} | {name} | {parent} | {wikidata} | {iptc_subject} |\n"
```

Comment:
it's more efficient to use a list and join it at the end than to extend a string.